packages/next/src/server/lib/mock-request.ts

```diff
@@ -56,6 +59,20 @@ export class MockedRequest extends Stream.Readable implements IncomingMessage {
     }
   }
 
+  public get headersDistinct(): NodeJS.Dict<string[]> {
+    const headers: NodeJS.Dict<string[]> = {}
+    for (const key in this.headers) {
+      const value = this.headers[key]
+      if (Array.isArray(value)) {
+        headers[key] = value
+      } else if (value) {
+        headers[key] = [value]
+      }
+    }
```

Comment:
Nit pick: remove a line if you use Object.entries()

```suggestion
    for (const [key, value] in Object.entries(this.headers)) {
      if (Array.isArray(value)) {
        headers[key] = value
      } else if (value) {
        headers[key] = [value]
      }
    }
```